Lib/unittest/suite.py

```diff
@@ -199,14 +207,25 @@ def _handleModuleFixture(self, test, result):
             try:
                 setUpModule()
             except Exception as e:
+                try:
+                    case.doModuleCleanups()
+                except Exception as exc:
+                    self._createClassOrModuleLevelException(result, exc,
+                                                            'setUpModule',
+                                                            currentModule)
                 if isinstance(result, _DebugResult):
                     raise
                 result._moduleSetUpFailed = True
-                errorName = 'setUpModule (%s)' % currentModule
-                self._addClassOrModuleLevelException(result, e, errorName)
+                self._createClassOrModuleLevelException(result, e,
+                                                        'setUpModule',
+                                                        currentModule)
             finally:
                 _call_if_exists(result, '_restoreStdout')
 
+    def _createClassOrModuleLevelException(self, result, exc, parent, name):
+        errorName = '{} ({})'.format(parent, name)
```

Comment:
This is Python 3.8, so what do you think about using `f{parent} ({name})`?

Reply:
Definitely! I wasn't sure how important maintaining the current style in the code is, I didn't want to jump the gun and make it 3.8 only if normally we try to keep the code more backwards compatible. I will move it to the f-string style :)